### ISSUE TITLE
[BUG] Adding new node to upgraded cluster causes upgrade job to be created for new node

### ISSUE THREAD

**Issue by Martin-Weiss** [harvester/harvester#2432]
**Describe the bug**
System Upgrade Controller creates a new job for a node that got added after upgrading existing Harvester nodes in the cluster.

**To Reproduce**
1. Install single node 1.0.0 cluster
2. Upgrade cluster to 1.0.1
3. Upgrade cluster to 1.0.2
4. Add two additional nodes to the cluster
5. See upgrade pods getting started for the new nodes even though they are on the correct version.

Logs in that pod:
```
Wait for upgrade repo ready...
+ curl -sfL http://upgrade-repo-hvst-upgrade-wmx6d.harvester-system/harvester-iso/harvester-release.yaml
+ echo 'Wait for upgrade repo ready...'
+ sleep 5
```

Assumption is that this happens, because the plans for the upgrades are still there and the new node does not have the annotations that are added to the nodes after the upgrade job had been completed.

**Expected behavior**
In case a node already has the correct version - no upgrade pod gets started - or even better - the upgrade pods exits successfully after it verified that the node already has the correct version.

**Environment**
 - Harvester ISO version:  see above
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): KVM nested virtualization for testing

**Comment by w13915984028** [comment]
@bk201 @starbops Please double check if it is still in v1.1.0, thanks.

**Comment by bk201** [comment]
@FrankYang0529 This is fixed in https://github.com/harvester/harvester/pull/2918. Please help add a test plan. Thanks!

**Comment by TachunLin** [comment]
Verified **fixed** on v1.1.0. Close this issue.

### Result
When one Harvester node upgrade from v1.0.1 -> v1.0.2 -> v1.0.3 to v1.1.0, we could able to add new nodes to the upgraded Harvester. 

* Successfully join the second node `harvester-node-1` after node 0 upgraded to v1.1.0
![image](https://user-images.githubusercontent.com/29251855/199696637-170d6f6f-34a3-4a74-8978-98684844b36b.png)

* Successfully join the third node `harvester-node-2` after node 0 upgraded to v1.1.0
![image](https://user-images.githubusercontent.com/29251855/199700231-30d2692d-8a7f-416a-a665-6d72be7efb73.png)


### Test Information
* Test Environment: 3 nodes harvester on local kvm machine
* Harvester version: `v1.1.0`

### Verify Steps
(refer to https://github.com/harvester/tests/issues/580#issue-1425241733)

1. Download https://github.com/harvester/ipxe-examples and checkout to v1.0 branch.
2. cd vagrant-pxe-harvester.
3. Update [harvester_cluster_nodes](https://github.com/harvester/ipxe-examples/blob/f58fc6f4f54866e318bb9519aae1644406225f45/vagrant-pxe-harvester/settings.yml#L25) to 1.
4. Update [the first node CPU](https://github.com/harvester/ipxe-examples/blob/f58fc6f4f54866e318bb9519aae1644406225f45/vagrant-pxe-harvester/settings.yml#L61) to 12.
5. Change [URL](https://github.com/harvester/ipxe-examples/blob/f58fc6f4f54866e318bb9519aae1644406225f45/vagrant-pxe-harvester/settings.yml#L15-L18) to harvester-v1.0.1.
6. Run `./setup_harvester.sh`
7. Upgrade node 1 from v1.0.1 -> v1.0.2 -> v1.0.3 until v1.1.0
![image](https://user-images.githubusercontent.com/29251855/199675459-747c04e4-09fc-417e-82be-0780d06fd477.png)
8. Checkout ipxe-eamples to main branch.
9. Login pxe_server and update some files.
```
vagrant ssh pxe_server
> sudo -i
> cd /var/www/harvester
```
10. Add `schema_version: 1` to config-join-1.yaml
11. Change network config to following
  ```
  management_interface:
      interfaces:
      - name: ens6   # The management interface name
      method: DHCP
  ```
12. cp config-join-1.yaml config-join-2.yaml
13. change `hostname` to `harvester-node-2` in `config-join-2.yaml`
14. chown www-data config-join-2.yaml
15. cp 02\:00\:00\:35\:86\:92 02\:00\:00\:2f\:f2\:2a
16. change `config-join-1.yaml` to `config-join-2.yaml` in `02\:00\:00\:2f\:f2\:2a`
17. update iso
```
> curl -o harvester-initrd-amd64 https://releases.rancher.com/harvester/v1.1.0/harvester-v1.1.0-initrd-amd64
> curl -o harvester-vmlinuz-amd64 https://releases.rancher.com/harvester/v1.1.0/harvester-v1.1.0-vmlinuz-amd64
> curl -o harvester-rootfs-amd64.squashfs https://releases.rancher.com/harvester/v1.1.0/harvester-v1.1.0-rootfs-amd64.squashfs
> curl -o harvester-amd64.iso https://releases.rancher.com/harvester/v1.1.0/harvester-v1.1.0-amd64.iso
```
18. Edit settings.yml 
19. Change harvester_cluster_nodes: 3
20. Update cpu and memory of the second and third node to proper size
21. Run vagrant up harvester-node-1 in ipxe-example folder 
22. Check can join the second node `harvester-node-1` after node 0 upgraded to v1.1.0
23. Run vagrant up harvester-node-2 in ipxe-example folder
